### PR TITLE
Cart feature: models and api routes

### PR DESCRIPTION
**Pull request by helenhisaac**
### Assignee Tasks

* [ ] added unit tests (or none needed)
* [ ] written relevant docs (or none needed)
* [ ] referenced any relevant issues (or none exist)

### Guidelines

Please add a description of this Pull Request's motivation, scope, outstanding issues or potential alternatives, reasoning behind the current solution, and any other relevant information for posterity.

---

_Your PR Notes Here_

### CODE REVIEW

File: server/db/models/cart.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const Sequelize = require('sequelize')
+const db = require('../db')
+
+const Cart = db.define('cart', {
+  candies: {
+    type: Sequelize.ARRAY(Sequelize.INTEGER),
```
Comment (erickasylvester) [inline server/db/models/cart.js]: it might be useful to add a note to state what the array of integers represent.

File: server/db/models/cart.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    defaultValue: []
+  }
+})
+
+Cart.prototype.remove = function(candy) {
+  this.candies = this.candies.filter(el => el !== candy)
+  return candy
```
Comment (erickasylvester) [inline server/db/models/cart.js]: Do we return the removed candy whether it was removed or not?

File: server/db/models/index.js [hunk 3]
```diff
@@ -8,6 +10,8 @@ const Candy = require('./candy')
  *    BlogPost.belongsTo(User)
  */
 
+// CartCandy.belongsTo(Cart)
```
Comment (erickasylvester) [inline server/db/models/index.js]: to be completed

File: server/db/models/cart.js (hunk 2 above)
Comment (helenhisaac) [reply server/db/models/cart.js]: Good point...I was debating with this whether I should actually just return the new candies array instead? And change the test to check that it doesn't include the candy id that was passed to the method.

**Pull request merged**